crypto/BRCryptoWalletManagerClient.c

```diff
@@ -626,20 +628,21 @@ cwmTransactionEventAsBTC (BRWalletManagerClientContext context,
             // ... update state to reflect included if the timestamp and block height are already set
             if (0 != btcTransaction->timestamp && TX_UNCONFIRMED != btcTransaction->blockHeight) {
 
+                BRCryptoTransferState oldState = cryptoTransferGetState (transfer);
+                assert (CRYPTO_TRANSFER_STATE_INCLUDED != oldState.type);
+
                 // The transfer is included and thus we now have a feeBasisConfirmed.  For BTC
                 // the feeBasisConfirmed is identical to feeBasisEstimated
                 BRCryptoFeeBasis feeBasisConfirmed = cryptoTransferGetEstimatedFeeBasis (transfer);
                 cryptoTransferSetConfirmedFeeBasis (transfer, feeBasisConfirmed);
+                BRCryptoAmount fee = cryptoFeeBasisGetFee (feeBasisConfirmed);
 
-                BRCryptoTransferState oldState = cryptoTransferGetState (transfer);
-                assert (CRYPTO_TRANSFER_STATE_INCLUDED != oldState.type);
-
-                // TODO(fix): BRCryptoTransferState leaks BRCryptoAmount; see CORE-484
                 BRCryptoTransferState newState = cryptoTransferStateIncludedInit (btcTransaction->blockHeight,
                                                                                   0,
                                                                                   btcTransaction->timestamp,
-                                                                                  cryptoFeeBasisGetFee (feeBasisConfirmed));
+                                                                                  fee);
```

Comment:
So this is in part stylistic - but pushing towards consistency in how we pass arguments (regarding reference counts).